packages/protocol/src/message-kinds/close.ts

```diff
@@ -8,7 +8,7 @@ import { Parser } from '../parser.js'
  */
 export type CreateCloseOptions = {
   data: CloseData
-  metadata: Omit<CloseMetadata, 'id' | 'kind' | 'createdAt'>
+  metadata: Omit<CloseMetadata, 'id' | 'kind' | 'createdAt' | 'protocol'> & { protocol?: CloseMetadata['protocol'] }
```

Comment:
Non-blocking: I kinda hate indexed access type syntax in typescript. Instead I prefer to just use the underlying type, in this case,
```
`${number}`
```
```suggestion
  metadata: Omit<CloseMetadata, 'id' | 'kind' | 'createdAt' | 'protocol'> & { protocol?: `${number}` }
```

Up to you if you feel like doing this. Just a personal preference on my part.

Reply:
Curious what you don't like about it! the way it looks, or is there something about the way it works that you dislike?

Reply:
it's a readability thing. it's an abstraction that i personally don't find helpful. i don't like needing to have to look at a separate file or type to figure out what i'm actually working with, especially if it's as simple as a number or string.

Reply:
Oh got it. I'm leaning towards leaving it as is because theoretically, the types could drift between `CloseMetadata` and this field if it's not referencing `CloseMetadata`.